delphi-frontend/src/main/java/au/com/integradev/delphi/preprocessor/directive/CompilerDirectiveParserImpl.java

```diff
@@ -321,11 +321,11 @@ private boolean isEndOfDirective(char character) {
     boolean result = false;
 
     if (directiveBracketType == CURLY) {
-      result = (character == '}');
+      result = character == '}' && peekChar() == END_OF_INPUT;
     }
 
     if (directiveBracketType == PAREN) {
-      result = (character == '*' && peekChar() == ')');
+      result = character == '*' && peekChar() == ')' && peekChar(1) == END_OF_INPUT;
```

Comment:
> I'm not sure why this is necessary. I reverted the change and all the tests passed.

Added the egregiously-named `CompilerDirectiveParserTest::testExpressionSurroundedWithNestedDirectivesShouldNotThrowException`

> As an aside, I'm not sure it's safe to represent `END_OF_INPUT` as `\0` since that's a valid whitespace character in Delphi.

This is a good point about the `END_OF_INPUT` sentinel value, but I'm not too bothered about it since this entire separate "compiler directive parser" should really disappear in the eventual future where a more authentic parsing solution is implemented with #261.

If it's a sticking point, we could look at using the same sentinel value as the ANTLR3 generated lexer (to at least minimize the number of magic character values)

Reply:
Honestly, anyone using a null byte as a whitespace character is asking for trouble. It's not a sticking point for me.